### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by tmigot**

### CODE REVIEW

File: docs/Project.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[deps]
+ADNLPModels = "54578032-b7ea-4c30-94aa-7cbd1cce6c9a"
+DCISolver = "bee2e536-65f6-11e9-3844-e5bb4c9c55c9"
```
Comment (abelsiqueira) [inline docs/Project.toml:3]:
```suggestion
```

File: docs/make.jl [hunk 2]
```diff
@@ -0,0 +1,27 @@
+using ADNLPModels
+using Documenter
+using Literate
```
Comment (abelsiqueira) [inline docs/make.jl:3]: Why do we need Literate?

File: docs/make.jl [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    format = Documenter.HTML(prettyurls = get(ENV, "CI", nothing) == "true"),
+    modules = [DCISolver],
+    pages = pages
+)
+
+# Documenter can also automatically deploy documentation to gh-pages.
+# See "Hosting Documentation" and deploydocs() in the Documenter manual
+# for more information.
+deploydocs(
+    repo = "github.com/JuliaSmoothOptimizers/DCISolver.jl.git"
```
Comment (abelsiqueira) [inline docs/make.jl:23]:
```suggestion
    repo = "github.com/JuliaSmoothOptimizers/DCISolver.jl.git",
    push_preview = true
```

File: docs/make.jl (hunk 2 above)
Comment (tmigot) [reply docs/make.jl]: I was first planning to use, and finally decided against it and forgot to remove it.
Thanks for the feedback !

File: docs/Project.toml (hunk 1 above)
Comment (tmigot) [reply docs/Project.toml]: thanks

File: docs/src/index.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# DCISolver - Dynamic Control of Infeasibility Solver
+
+DCI is a solver for equality-constrained nonlinear problems, i.e.,
+optimization problems of the form
+
+```math
+    min_x f(x) \quad \text{s.t.} \quad  c(x) = 0,
```
Comment (abelsiqueira) [inline docs/src/index.md:7]:
```suggestion
    \min_x \ f(x) \quad \text{s.t.} \quad  c(x) = 0,
```

File: docs/src/index.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+## Example
+
+```@example
+using DCISolver, ADNLPModels
+nlp = ADNLPModel(x -> 100 * (x[2] - x[1]^2)^2 + (x[1] - 1)^2, 
+                 [-1.2; 1.0],
+                 x -> [x[1] * x[2] - 1], 
+                 [0.0], [0.0],
+                 name = "Rosenbrock with x₁x₂=1")
```
Comment (abelsiqueira) [inline docs/src/index.md:38]:
```suggestion
nlp = ADNLPModel(
  x -> 100 * (x[2] - x[1]^2)^2 + (x[1] - 1)^2, 
  [-1.2; 1.0],
  x -> [x[1] * x[2] - 1], 
  [0.0], [0.0],
  name = "Rosenbrock with x₁x₂=1"
)
```

File: docs/src/index.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+stats = dci(nlp, nlp.meta.x0)
```
Comment (abelsiqueira) [inline docs/src/index.md:39]:
```suggestion
stats = with_logger(NullLogger()) do
  dci(nlp, nlp.meta.x0)
end
```

File: docs/src/index.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+```
+
+It uses [LDLFactorizations.jl](https://github.com/JuliaSmoothOptimizers/LDLFactorizations.jl) by default to compute the factorization in the tangent step. Follow [HSL.jl](https://github.com/JuliaSmoothOptimizers/HSL.jl)'s `MA57` installation for an alternative.
+
+The feasibility steps are factorization-free and use iterative methods from [Krylov.jl](https://github.com/JuliaSmoothOptimizers/Krylov.jl).
+
+## Example
+
+```@example
+using DCISolver, ADNLPModels
```
Comment (abelsiqueira) [inline docs/src/index.md:33]:
```suggestion
using DCISolver, ADNLPModels, Logging
```

File: docs/src/fine-tuneDCI.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+m, n = nlp.meta.ncon, nlp.meta.nvar
+meta = DCISolver.MetaDCI(nlp.meta.x0, nlp.meta.y0, max_time = 600., 
+                                                   linear_solver = :ldlfact, 
+                                                   TR_compute_step = :TR_lsmr)
+stats = dci(nlp, nlp.meta.x0, meta)
+
+#The alternative would be:
+stats2 = dci(nlp, nlp.meta.x0, max_time = 600., 
+                               linear_solver = :ldlfact, 
+                               TR_compute_step = :TR_lsmr)
```
Comment (abelsiqueira) [inline docs/src/fine-tuneDCI.md:40]: Reformat and adjust like the code in Introduction

File: docs/src/fine-tuneDCI.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    
+    #Solver for the factorization
+    linear_solver :: Symbol # = :ldlfact,#:ma57,
+
+  #Normal step
+    feas_step :: Symbol #:feasibility_step
+  #Feasibility step in the normal step
+    TR_compute_step :: Symbol #:TR_lsmr, :TR_dogleg
+    TR_compute_step_struct :: Union{TR_lsmr_struct, TR_dogleg_struct}
+
```
Comment (abelsiqueira) [inline docs/src/fine-tuneDCI.md:70]: Fix the indentation

File: docs/src/fine-tuneDCI.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+#dci function call:
+    #Tolerances on the problem:
+    atol :: AbstractFloat # = 1e-5,
+    rtol :: AbstractFloat # = 1e-5, #ϵd = atol + rtol * dualnorm
+    ctol :: AbstractFloat # = 1e-5, #feasibility tolerance
+
+    #Evaluation limits
+    max_eval :: Integer # = 50000,
+    max_time :: AbstractFloat # = 120.
+    max_iter :: Integer #:: Int = 500
```
Comment (abelsiqueira) [inline docs/src/fine-tuneDCI.md:56]:
```suggestion
    max_iter :: Integer # = 500
```

File: docs/src/fine-tuneDCI.md (hunk 9 above)
Comment (abelsiqueira) [reply docs/src/fine-tuneDCI.md:70]: Instead of writing `# = ...`, say `# default: ...
Leave a space after `#`.

File: docs/src/fine-tuneDCI.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+    #Compute Lagrange multipliers
+    comp_λ :: Symbol # :cgls
+    λ_struct :: comp_λ_cgls # comp_λ_cgls(length(x0), length(y0), atol)
+    
+    #Solver for the factorization
+    linear_solver :: Symbol # = :ldlfact,#:ma57,
```
Comment (abelsiqueira) [inline docs/src/fine-tuneDCI.md:63]:
```suggestion
    linear_solver :: Symbol # default: :ldlfact, options: :ldlfact, :ma57.
```

**Review by abelsiqueira — CHANGES_REQUESTED** [review]
I was hoping that `push_preview` would preview the docs, but only from inside the docs, so I had to build locally anyway.
I made another pass, probably the last.
@dpo, do you want to take a look or can I merge afterwards?

**Comment by tmigot** [comment]
> I was hoping that `push_preview` would preview the docs, but only from inside the docs, so I had to build locally anyway.
> I made another pass, probably the last.

Thanks for the suggestion, it's much better now! 
Unfortunately, I don't know a simple way to visualize working docs.

**Comment by abelsiqueira** [comment]
> Thanks for the suggestion, it's much better now!
> Unfortunately, I don't know a simple way to visualize working docs.

Do you mean online or locally? Online, you would have to set up doc building for different branches and add an SSH key to your repository, so it's too much work.

File: docs/make.jl [hunk 12]
```diff
@@ -0,0 +1,24 @@
+using ADNLPModels
+using Documenter
+using Printf
+using DCISolver
+
+# Add index.md file as introduction to navigation menu
+pages = ["Introduction"=> "index.md",
```
Comment (abelsiqueira) [inline docs/make.jl:7]:
```suggestion
pages = ["Introduction" => "index.md",
```

File: docs/make.jl [hunk 13]
```diff
@@ -0,0 +1,24 @@
+using ADNLPModels
+using Documenter
+using Printf
+using DCISolver
+
+# Add index.md file as introduction to navigation menu
```
Comment (abelsiqueira) [inline docs/make.jl:6]:
```suggestion
```

File: docs/make.jl [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+makedocs(
+    sitename = "DCISolver.jl",
+    format = Documenter.HTML(prettyurls = get(ENV, "CI", nothing) == "true"),
+    modules = [DCISolver],
+    pages = pages
+)
+
+# Documenter can also automatically deploy documentation to gh-pages.
+# See "Hosting Documentation" and deploydocs() in the Documenter manual
+# for more information.
```
Comment (abelsiqueira) [inline docs/make.jl:20]:
```suggestion
```

**Comment by abelsiqueira** [comment]
Thanks, and sorry, I forgot about this.

**Pull request merged**